File: scripts/projects_config.py
"""Managed project definitions for grimoire CLI."""

from __future__ import annotations

from dataclasses import dataclass, field
from pathlib import Path

GITHUB = Path.home() / "github"


@dataclass
class ManagedProject:
    name: str
    path: Path
    install_command: list[str]
    verify_commands: list[list[str]] = field(default_factory=list)


MANAGED_PROJECTS: list[ManagedProject] = [
    # --- TypeScript Web Apps ---
    ManagedProject(
        name="podwatch",
        path=GITHUB / "podwatch",
        install_command=["bun", "install"],
        verify_commands=[["bun", "run", "lint"], ["bun", "run", "typecheck"]],
    ),
    ManagedProject(
        name="rip",
        path=GITHUB / "rip",
        install_command=["bun", "install"],
        verify_commands=[["bun", "run", "lint"], ["bun", "run", "typecheck"]],
    ),
    # --- Chess Platform ---
    ManagedProject(
        name="elchess",
        path=GITHUB / "elchess",
        install_command=["bun", "install"],
        verify_commands=[["bun", "run", "lint"], ["bun", "run", "typecheck"]],
    ),
    # --- Mobile ---
    ManagedProject(
        name="CallRift",
        path=GITHUB / "CallRift",
        install_command=["bun", "install"],
        verify_commands=[["bun", "run", "lint"], ["bun", "run", "typecheck"]],
    ),
    # --- CI Pipeline ---
    ManagedProject(
        name="pr-firefighter",
        path=GITHUB / "pr-firefighter",
        install_command=["bun", "install"],
        verify_commands=[["bun", "run", "lint"], ["bun", "run", "typecheck"]],
    ),
    # --- Ops CLI ---
    ManagedProject(
        name="grimoire",
        path=GITHUB / "grimoire",
        install_command=["bun", "install"],
        verify_commands=[["bun", "run", "lint"], ["bun", "run", "typecheck"]],
    ),
    # --- Content ---
    ManagedProject(
        name="Sawyer-Visual-Media",
        path=GITHUB / "Sawyer-Visual-Media",
        install_command=["bun", "install"],
        verify_commands=[],
    ),
    # --- Python ---
    ManagedProject(
        name="scripts",
        path=GITHUB / "scripts",
        install_command=["echo", "no install needed"],
        verify_commands=[["ruff", "check", "."]],
    ),
    ManagedProject(
        name="augur",
        path=GITHUB / "augur",
        install_command=["uv", "sync"],
        verify_commands=[
            ["uv", "run", "ruff", "check", "."],
            ["uv", "run", "mypy", "."],
        ],
    ),
]

File: scripts/projects-config.ts
import { resolve } from "node:path";

export type ManagedProject = {
  installCommand: string[];
  name: string;
  path: string;
  verifyCommands: string[][];
};

const home = process.env.HOME ?? "/home/sawyer";
const githubRoot = process.env.GITHUB_ROOT
  ? resolve(process.env.GITHUB_ROOT)
  : resolve(home, "github");

export const managedProjects: ManagedProject[] = [
  {
    installCommand: ["bun", "install"],
    name: "next-web-template",
    path: resolve(githubRoot, "next-web-template"),
    verifyCommands: [
      ["bun", "run", "lint"],
      ["bun", "run", "typecheck"],
      ["bun", "run", "build"],
    ],
  },
  {
    installCommand: ["bun", "install"],
    name: "next-blog-template",
    path: resolve(githubRoot, "next-blog-template"),
    verifyCommands: [
      ["bun", "run", "lint"],
      ["bun", "run", "typecheck"],
      ["bun", "run", "build"],
    ],
  },
];
